includes/modules/pages/product_info/main_template_vars.php

```diff
@@ -22,7 +22,7 @@
   $product_not_found = $product_info->EOF;
 
   if (!defined('PRODUCT_THROWS_200_WHEN_DISABLED') || PRODUCT_THROWS_200_WHEN_DISABLED !== true) {
-      if ($product_info->fields['products_status'] != 1) {
+      if (!$product_not_found && $product_info->fields['products_status'] != 1) {

```

Comment:
Why this addition?

Reply:
> Why this addition?

If the product does not exist in the database, then using any specific field from the query provides a notice. Therefore, to omit the notice some sort of test against data identifying the absence of the product is needed. 

Seeing that the next line was intended to cause the variable `$product_not_found` to go true where it's default is false offered an opportunity to reuse that variable. At this point in the code, it seems that it has already been identified that the product does not exist at all in the database.

If there is something additional expected to occur in that section (regardless of the existence of the product in the database) then perhaps something similar to the test in the previous commit can be reincorporated. i.e. `!isset($product_info->fields['products_status']) ||`.

There seems to be a lot of code consolidation going on though where reused "long" variables are being "shortened" or assigned just one time instead. In the current state, that's what seemed to be possible here.